kaldb/src/test/java/com/slack/kaldb/server/ManagerApiGrpcTest.java

```diff
@@ -0,0 +1,409 @@
+            ManagerApi.GetServiceMetadataRequest.newBuilder().setName(serviceName).build());
+
+    assertThat(secondAssignment.getThroughputBytes()).isEqualTo(throughputBytes);
+    assertThat(secondAssignment.getPartitionConfigsList().size()).isEqualTo(2);
+    assertThat(secondAssignment.getPartitionConfigsList().get(0).getPartitionsList())
+        .isEqualTo(List.of("1", "2"));
+    assertThat(secondAssignment.getPartitionConfigsList().get(0).getEndTimeEpochMs())
+        .isNotEqualTo(Long.MAX_VALUE);
+
+    assertThat(secondAssignment.getPartitionConfigsList().get(1).getPartitionsList())
+        .isEqualTo(List.of("3", "4", "5"));
+    assertThat(secondAssignment.getPartitionConfigsList().get(1).getStartTimeEpochMs())
+        .isGreaterThanOrEqualTo(nowMs);
+    assertThat(secondAssignment.getPartitionConfigsList().get(1).getEndTimeEpochMs())
+        .isEqualTo(Long.MAX_VALUE);
+
+    // only update the throughput, leaving the partition assignments
+    long updatedThroughputBytes = 12;
+    managerApiStub.updatePartitionAssignment(
+        ManagerApi.UpdatePartitionAssignmentRequest.newBuilder()
+            .setName(serviceName)
+            .setThroughputBytes(updatedThroughputBytes)
+            .build());
+    Metadata.ServiceMetadata thirdAssignment =
+        managerApiStub.getServiceMetadata(
+            ManagerApi.GetServiceMetadataRequest.newBuilder().setName(serviceName).build());
+
+    assertThat(thirdAssignment.getThroughputBytes()).isEqualTo(updatedThroughputBytes);
+    assertThat(thirdAssignment.getPartitionConfigsList().size()).isEqualTo(2);
+    assertThat(thirdAssignment.getPartitionConfigsList().get(0).getPartitionsList())
+        .isEqualTo(List.of("1", "2"));
+    assertThat(thirdAssignment.getPartitionConfigsList().get(0).getEndTimeEpochMs())
+        .isNotEqualTo(Long.MAX_VALUE);
+
+    assertThat(thirdAssignment.getPartitionConfigsList().get(1).getPartitionsList())
+        .isEqualTo(List.of("3", "4", "5"));
+    assertThat(thirdAssignment.getPartitionConfigsList().get(1).getStartTimeEpochMs())
+        .isGreaterThanOrEqualTo(nowMs);
+    assertThat(thirdAssignment.getPartitionConfigsList().get(1).getEndTimeEpochMs())
+        .isEqualTo(Long.MAX_VALUE);
+  }
+
+  @Test
+  public void shouldErrorUpdatingPartitionAssignmentNonexistentService() {
+    String serviceName = "testService";
+    List<String> partitionList = List.of("1", "2");
+
+    StatusRuntimeException throwable1 =
+        (StatusRuntimeException)
+            catchThrowable(
+                () ->
+                    managerApiStub.updatePartitionAssignment(
+                        ManagerApi.UpdatePartitionAssignmentRequest.newBuilder()
+                            .setName(serviceName)
+                            .setThroughputBytes(-1)
+                            .addAllPartitionIds(partitionList)
+                            .build()));
+    assertThat(throwable1.getStatus().getCode()).isEqualTo(Status.UNKNOWN.getCode());
```

Comment:
assertThatExceptionIsThrownBy() may be a simpler way to assert this condition.

Reply:
I initially had that approach but wanted to also assert specific GRPC codes and messages. In order to this, I needed access to the throwable so I could access the getters for the methods required.